domdiv/draw.py

```diff
@@ -22,15 +23,337 @@ def split(l, n):
+        canvas.translate(x, y)
+        canvas.rotate(rotation)
+
+    def translateCropmarkEnable(self, side):
+        # Returns True if a cropmark is needed on that side of the card
+        # Takes into account the card's rotation, if the tab is flipped, if the card is next to an edge, etc.
+
+        # First the rotation. The page does not change even if the card is rotated.
+        # So need to translate page side to the actual drawn card edge
+        if self.rotation == 0:
+            sideTop = self.cropOnTop
+            sideBottom = self.cropOnBottom
+            sideRight = self.cropOnRight
+            sideLeft = self.cropOnLeft
+        elif self.rotation == 90:
+            sideTop = self.cropOnRight
+            sideBottom = self.cropOnLeft
+            sideRight = self.cropOnBottom
+            sideLeft = self.cropOnTop
+        elif self.rotation == 180:
+            sideTop = self.cropOnBottom
+            sideBottom = self.cropOnTop
+            sideRight = self.cropOnLeft
+            sideLeft = self.cropOnRight
+        elif self.rotation == 270:
+            sideTop = self.cropOnLeft
+            sideBottom = self.cropOnRight
+            sideRight = self.cropOnTop
+            sideLeft = self.cropOnBottom
+
+        # Now can return the proper value based upon what side is requested
+        if side == self.TOP:
+            return sideTop
+        elif side == self.BOTTOM:
+            return sideBottom
+        elif side == self.RIGHT:
+            return sideRight
+        elif side == self.LEFT:
+            return sideLeft
+        else:
+            return False  # just in case
+
+
+class Plotter(object):
+    # Creates a simple plotting object that goes from point to point.
+    # This makes outline drawing easier since calculations only need to be the delta from
+    # one point to the next.  The default plotting in reportlab requires both
+    # ends of the line in absolute sense.  Thus calculations can become increasingly more
+    # complicated given various options.  Using this object simplifies the calculations significantly.
+
+    def __init__(self, canvas, x=0, y=0, cropmarkLength=-1, cropmarkSpacing=-1):
+        self.canvas = canvas
+        self.x = x
+        self.y = y
+        self.LEFT, self.RIGHT, self.TOP, self.BOTTOM, self.LINE, self.NO_LINE, self.DOT = range(1, 8)  # Constants
+        if cropmarkLength < 0:
+            cropmarkLength = 0.2
+        if cropmarkSpacing < 0:
+            cropmarkSpacing = 0.1
+        self.CropMarkLength = cropmarkLength * cm  # The length of a cropmark
+        self.CropMarkSpacing = cropmarkSpacing * cm  # The spacing between the cut point and the start of the cropmark
+        self.DotSize = 0.2  # Size of dot marks
+
+    def setXY(self, x, y):
+        self.x = x
+        self.y = y
+
+    def getXY(self):
+        return (self.x, self.y)
+
+    def move(self, delta_x=0, delta_y=0, pen=False):
```

Comment:
maybe `plot` as it's actually drawing, not just moving?

Reply:
For now, I'm keeping, since the move can be with pen up or pen down.